### PR TITLE
MGMT-10493: Don't truncate CAPI/Hypershift logs in CI

### PR DESCRIPTION
**Pull request by mkowalski**
## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [x] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @omertuc 
/cc @osherdp 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: deploy/operator/gather.sh [hunk 1]
```diff
@@ -129,15 +129,15 @@ function gather_capi_data() {
     done
   done
 
-  oc logs deployment/operator -n hypershift > "${capi_dir}/hypershift.log" || true
+  oc logs --tail=-1 deployment/operator -n hypershift > "${capi_dir}/hypershift.log" || true
```
Comment (osherdp) [inline deploy/operator/gather.sh:132]: are you sure it currently truncates?
so far we've only seen the need of this flag when using ``--selector``

File: deploy/operator/gather.sh [hunk 2]
```diff
@@ -129,15 +129,15 @@ function gather_capi_data() {
...
 
   # Detect namespace where the capi-provider lives. The specific name depends whether deployed standalone or via
   # Hypershift, that's why we grep for both.
   capi_provider_ns=$(oc get pods --no-headers -A -o jsonpath='{range .items[*]}{@.metadata.name}{" "}{@.metadata.namespace}{"\n"}' | egrep "capi-provider|cluster-api-provider-agent" | awk -F " " '{print $2}' | head -n1) || true
   capi_provider_dir="${capi_dir}/${capi_provider_ns}"
   mkdir -p "${capi_provider_dir}"
 
-  oc get pods -n ${capi_provider_ns} -o=custom-columns=NAME:.metadata.name --no-headers | xargs -r -I {} sh -c "oc logs {} -n ${capi_provider_ns} --all-containers > ${capi_provider_dir}/logs_{}.log" || true
+  oc get pods -n ${capi_provider_ns} -o=custom-columns=NAME:.metadata.name --no-headers | xargs -r -I {} sh -c "oc logs --tail=-1 {} -n ${capi_provider_ns} --all-containers > ${capi_provider_dir}/logs_{}.log" || true
```
Comment (osherdp) [inline deploy/operator/gather.sh:140]: same question. I don't see any proof for it to be truncated:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_assisted-service/3989/pull-ci-openshift-assisted-service-master-edge-e2e-metal-assisted-operator-capi/1539256049951838208/artifacts/e2e-metal-assisted-operator-capi/assisted-baremetal-operator-gather/artifacts/capi/assisted-spoke-cluster-assisted-test-cluster/

**Review by osherdp — CHANGES_REQUESTED** [review]

File: deploy/operator/gather.sh (hunk 1 above)
Comment (mkowalski) [reply deploy/operator/gather.sh:132]: It is only to be on the safe side. The current code does not cause any problems. But also to be honest, the test where it runs is a fairly simple one so we never expect there to be a flood of logs. In the end feel free to push it or hold as you wish. From my perspective (as someone who just needs access to CAPI logs) it works as it is currently

**Pull request merged**